gateway-ha/src/test/java/io/trino/gateway/ha/TestGatewayHaMultipleBackend.java

```diff
@@ -359,6 +360,30 @@ void testCookieSigning()
         assertThat(callbackResponse.code()).isEqualTo(500);
     }
 
+    @Test
+    void testHealthCheckEndpoints()
+            throws IOException
+    {
+        Request livenessCheck = new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/trino-gateway/livez")
+                        .build();
+        Response livenessResponse = httpClient.newCall(livenessCheck).execute();
+        assertThat(livenessResponse.code()).isEqualTo(200);
+
+        Request readinessCheck = new Request.Builder()
+                .url("http://localhost:" + routerPort + "/trino-gateway/readyz")
+                .build();
+        for (int i = 0; i < 100; i++) {
+            try (Response readinessResponse = httpClient.newCall(readinessCheck).execute()) {
+                if (readinessResponse.code() == 200) {
+                    return;
+                }
+            }
+            sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
```

Comment:
Why not a regular sleep?

Reply:
I just refer to the exist code: https://github.com/trinodb/trino-gateway/blob/379ca35c2b4fab28cb58b58d281b97f85bdc31a0/gateway-ha/src/test/java/io/trino/gateway/ha/HaGatewayTestUtils.java#L168
That shouldn't make any practical difference.